testsplitter.py

```diff
@@ -79,20 +92,43 @@ def initUI(self):
         self.setWindowTitle('QSplitter demo')
         self.show()
 
+    def sliderCreation(self, min, max, value, interval):
+        """define a slider with default values if parameter are not set"""
+        self.slider = QSlider(Qt.Horizontal)
+        self.slider.setMinimum(min)
+        self.slider.setMaximum(max)
+        self.slider.setValue(value)
+        self.slider.setTickPosition(QSlider.TicksBelow)
+        self.slider.setTickInterval(interval)
+        self.slider.valueChanged.connect(self.valueChange)
+        return self.slider
+
+    def valueChange(self):
+        sliderCoef = self.slider.value()
+        self.textedit.append("inside of slider: " + str(sliderCoef))
+        return sliderCoef
+
+
     def finalSize(self, objectSize):
         """will define the new size of object in regard of """
-        if self.halfSize.isChecked() == True:
-            coef = 0.5
-        if self.normalSize.isChecked() == True:
-            coef = 1
-        if self.doubleSize.isChecked() == True:
-            coef = 2
+        # list will be used to copy the list and not to refer at the same list
+        modifiedSize = list(objectSize)
+
+        # self.textedit.append("value of slider bar in final size" + str(userCoef))
+        # self.textedit.append("value of slider bar in final size" + str(self.valueChange()))
+        # temporary removed
+        #if self.halfSize.isChecked() == True:
+        #    coef = 0.5
+        #if self.normalSize.isChecked() == True:
+        #    coef = 1
+        #if self.doubleSize.isChecked() == True:
+        #    coef = 2
         try:
             for i in xrange(len(objectSize)):
-                objectSize[i]= objectSize[i]*coef
-            return objectSize
+                modifiedSize[i]= modifiedSize[i] * (self.valueChange())
```

Comment:
Use list comprehension
```
vc = self.valueChange()
modifiedSize [ o * vc for o in objectSize]
```
That's 2 loops removed, and useless recall of self.valueChanged() done only once